components/fxa-client/src/lib.rs

```diff
@@ -104,7 +104,9 @@ pub enum FxaError {
 /// account and accessing other services on behalf of the user.
 ///
 pub struct FirefoxAccount {
-    internal: internal::FirefoxAccount,
+    // For now, we serialize all access on a single `Mutex` for thread safety across
+    // the FFI. We should make the locking more granular in future.
+    internal: std::sync::Mutex<internal::FirefoxAccount>,
```

Comment:
Right now I am very happy that I went with this `internal` abstraction that could be easily wrapped in a single Mutex 🎉

Reply:
(For context: the previous version of UniFFI was adding a Mutex like this automagically, but that behaviour is deprecated and will be removed in a future version)